src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -82,46 +175,18 @@ public Driver(DriverSettings settings, IEntryPoint<TIn, TOut> entryPoint)
         /// <returns>The exit code.</returns>
         public async Task<int> Run(string[] args)
         {
-            var simulate = new Command("simulate", "(default) Run the program using a local simulator.")
-            {
-                Handler = CommandHandler.Create<ParseResult, string>(Simulate)
-            };
-            AddOptionIfAvailable(simulate, SimulatorOption);
+            var simulateSubCommands = this.entryPoints.Select(this.CreateSimulateEntryPointCommand).ToList();
+            var submitSubCommands = this.entryPoints.Select(this.CreateSubmitEntryPointCommand).ToList();
 
-            var submit = new Command("submit", "Submit the program to Azure Quantum.")
-            {
-                IsHidden = true,
-                Handler = CommandHandler.Create<ParseResult, AzureSettings>(Submit)
-            };
-            AddOptionIfAvailable(submit, SubscriptionOption);
-            AddOptionIfAvailable(submit, ResourceGroupOption);
-            AddOptionIfAvailable(submit, WorkspaceOption);
-            AddOptionIfAvailable(submit, TargetOption);
-            AddOptionIfAvailable(submit, StorageOption);
-            AddOptionIfAvailable(submit, AadTokenOption);
-            AddOptionIfAvailable(submit, BaseUriOption);
-            AddOptionIfAvailable(submit, LocationOption);
-            AddOptionIfAvailable(submit, JobNameOption);
-            AddOptionIfAvailable(submit, ShotsOption);
-            AddOptionIfAvailable(submit, OutputOption);
-            AddOptionIfAvailable(submit, DryRunOption);
-            AddOptionIfAvailable(submit, VerboseOption);
-            MarkOptionsAsMutuallyExclusive(
-                submit, 
-                new[] { BaseUriOption.Aliases.First(), LocationOption.Aliases.First() });
-
-            var root = new RootCommand(entryPoint.Summary) { simulate, submit };
-            foreach (var option in entryPoint.Options)
-            {
-                root.AddGlobalOption(option);
-            }
+            var (simulate, simulateValidators) = CreateSimulateCommand(simulateSubCommands);
+            var (submit, _) = CreateSubmitCommand(submitSubCommands);
```

Comment:
If the `submit` validators aren't used, could `CreateSubmitCommand` return only a `Command`?

Reply:
Tailoring the function to its specific use here by removing the validator part of the returned tuple undermines the function's purpose and degrades modularity of the function. The function may add validators to the command it creates, so those validators should be returned with the command, even if they aren't used in this particular case.

Reply:
OK.